osu.Game.Modes.Osu/Objects/Drawables/DrawableSlider.cs

```diff
@@ -107,7 +107,7 @@ protected override void Update()
 
             if (repeat > currentRepeat)
             {
-                if (ball.Tracking)
+                if (repeat < slider.RepeatCount && ball.Tracking)

```

Comment:
Why don't we just let this code take care of the last hitsound and instead don't manually invoke it a second time at the other place?